mabl snippets/isElementPresent.js

```diff
@@ -1,26 +1,28 @@
 /**
  * This snippet will allow you to assert against an element 
- * as long as the display of the element changes from 'none' 
- * to anything else.
+ * as long as the element is visible.
  * @param {object} - mablInputs Object containing input
  *                   variables (mablInputs.variables.user)
  * @param {function} callback - The callback function
  */
 function mablJavaScriptStep(mablInputs, callback) {
 
-    // this is where you would define what your element is
-    let element = document.querySelector('REPLACE_WITH_CSS_SELECTOR');
-    
-    // conditional statement to assure that the element is present, as well as
-    // assuring the display does not equal 'none'
-    if (element) {
-        const style = window.getComputedStyle(element);
-        if (style && style.display !== 'none' && style.visibility !== 'collapse' && style.visibility !== 'hidden') {
-             callback('Element Present');
-        } else {
-        throw Error('Element Not Visible');
-        }
-     } else {
-        throw Error('Element Not Present');
-    }
+  // this is where you would define what your element is
+  let element = document.querySelector('REPLACE_WITH_CSS_SELECTOR');
+
+  // conditional statement to assure that the element is present, as well as
+  // assuring the display does not equal 'none'
+  if (!element) {
+    throw Error('Element Not Present');
+  }
+
+  // check display and visibility values to determine if element is present. Can replace this with "or" statements
+  // in between, or even get rid of certain CSS computed style checks to match your application's implementation
+  const style = getComputedStyle(element);
+  if (style && style.display == 'none' && style.visibility == 'collapse' && style.visibility == 'hidden') {
```

Comment:
i think the logic you want is `if(style && (style.display === 'none' || style.visibility === 'collapse' || style.visibility === 'hidden')) {`.  With the whole thing as a boolean conjunction you couldn't ever match this condition since you're asserting that visibility has to have two different values at the same time.

Reply:
I was thinking the same, will make that change